src/components/browse/components/FacetList/FacetOfFacets.js

```diff
@@ -2,29 +2,69 @@
 
 import React from 'react';
 import PropTypes from 'prop-types';
-
+import memoize from 'memoize-one';
+import ReactTooltip from 'react-tooltip';
 import { Collapse } from './../../../ui/Collapse';
 import { Fade } from './../../../ui/Fade';
 
 /**
  * Used to render individual facet fields and their available terms in FacetList.
- *
- * @memberof module:facetlist
- * @class Facet
- * @type {Component}
  */
 export class FacetOfFacets extends React.PureComponent {
+
+
+    static anyFacetsHaveSelection(renderedFacets){
+        for (let facetIdx = 0; facetIdx < renderedFacets.length; facetIdx++){
+            const renderedFacet = renderedFacets[facetIdx]; // We have rendered facets as `props.facets`
+            const { anyTermsSelected: anySelected } = renderedFacet.props;
+            if (anySelected) {
+                return true;
+            }
+        }
+        return false;
+    }
+
     constructor(props){
         super(props);
         this.handleOpenToggleClick = this.handleOpenToggleClick.bind(this);
         this.handleExpandListToggleClick = this.handleExpandListToggleClick.bind(this);
+
+        this.memoized = {
+            anyFacetsHaveSelection: memoize(FacetOfFacets.anyFacetsHaveSelection)
+        };
+
+        // Most of this logic (facetOpen/facetClosing at least) is same between this and FacetTermsList.
+        // Maybe we could pull it out into reusable controller component. Maybe. Very low priority.
         this.state = {
-            'facetOpen'     : typeof props.defaultFacetOpen === 'boolean' ? props.defaultFacetOpen : true,
+            'facetOpen'     : typeof props.defaultGroupOpen === 'boolean' ? props.defaultGroupOpen : true,
             'facetClosing'  : false,
             'expanded'      : false
         };
     }
 
+    componentDidUpdate(pastProps, pastState){
```

Comment:
Initially, `mounted` is always false (due to serverside render) and is used in `renderFacet` to help set `defaultFacetOpen`(and then `defaultGroupOpen`, based on that) which transitively are always false as well on initial render. We thus need this `componentDidUpdate` to pickup when FacetList has mounted.

Reply:
Most of this logic is exact same as in FacetTermsList, just using defaultGroupOpen instead.

Reply:
It's mostly onMount because windowHeight is used to calculate `maxTermsToShow` sometimes; a transition from closed->open then occurs onMount/didUpdate>mountChanged for those which should be open. Theoretically not too important & can hardcode `maxTermsToShow` value to maybe get simpler code + performance (no transition). (idk)